### PR TITLE
Bazel clean before all post-submit jobs.

### PR DESCRIPTION
**Pull request by spxtr**
This has a few nice benefits.

### CODE REVIEW

File: jobs/ci-test-infra-bazel.sh [hunk 1]
```diff
@@ -23,10 +23,9 @@ pip install -r jenkins/test_history/requirements.txt
 # Cache location.
 export TEST_TMPDIR="/root/.cache/bazel"
 
-bazel build //... && rc=$? || rc=$?
+bazel clean
```
Comment (mikedanese) [inline jobs/ci-test-infra-bazel.sh]: I don't think bazel clean clears external repos.

Comment (spxtr) [inline jobs/ci-test-infra-bazel.sh]: Fixed.

**Comment by spxtr** [comment]
@mikedanese PTAL

**Comment by fejta** [comment]
What are these nice benefits?

**Comment by ixdy** [comment]
possibly "runs all tests", per https://github.com/kubernetes/kubernetes/issues/41391#issuecomment-309845173.

there's a better way to achieve that, though.

**Pull request merged**